### PR TITLE
client: fix client forever loading when connecting to off server

### LINKED ISSUE #1495: client: web UI shows markets progress bar everywhere when not conneted to a dex.
Starting with a dexc that is offline. Start it up with the dex it was pointing to down. You get the following progress bar on top of everything, making everything unclickable without a hard refresh of the page. On Firefox at least.

I think the progress bar is only intended to be on the markets for that dex.

image

If I want to view past orders, I get the progress bar and a js error:
```
Uncaught (in promise) TypeError: e is undefined
    Fr entry.js:1
    h markets.js:121
    F markets.js:120
    value app.js:208
    i app.js:194
    Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    Ho entry.js:1
    Ho entry.js:1
    value entry.js:1
    value app.js:359
    value doc.js:87
    value app.js:357
    value app.js:346
    value app.js:407
    l app.js:127
    Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    Ho entry.js:1
    Ho entry.js:1
    value entry.js:1
    <anonymous> index.js:7
    <anonymous> entry.js:1
[entry.js:1:120126](http://127.0.0.3:5758/js/entry.js?v=zA3PB0J)
```

If I want to configure my wallets I get the progress bar and the js error:
```
Uncaught (in promise) TypeError: can't convert null to object
    b wallets.js:210
    cn entry.js:1
    a entry.js:1
    promise callback*cn entry.js:1
    a entry.js:1
    un entry.js:1
    un entry.js:1
    value entry.js:1
    E wallets.js:147
    value app.js:208
    i app.js:194
    Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    Ho entry.js:1
    Ho entry.js:1
    value entry.js:1
    value app.js:359
    value doc.js:87
    value app.js:357
    value app.js:346
    value app.js:407
    l app.js:127
    Vo entry.js:1
    a entry.js:1
    promise callback*Vo entry.js:1
    a entry.js:1
    Ho entry.js:1
    Ho entry.js:1
    value entry.js:1
    <anonymous> index.js:7
    <anonymous> entry.js:1
[wallets.js:210:34](webpack://dexclient/src/js/wallets.js)
```

I don't think the js errors are related.

Everything else gives the progress bar. Dexc will give an error `2022-02-28 17:51:36.031 [ERR] CORE[wss://127.0.0.1:17273/ws]: (*wsConn).Request(route 'connect') Send error (cannot send on a broken connection), unregistering msg ID 1 handler`

The pages are accessible if you go to a page and do a refresh. If you go back to the markets page, the progress bar returns.

### PR DESCRIPTION
**Pull request by vctt94**
closes #1495

### CODE REVIEW

File: client/webserver/site/src/js/markets.ts [hunk 1]
```diff
@@ -2228,7 +2228,7 @@ class MarketList {
   /* exists will be true if the specified market exists. */
   exists (host: string, baseID: number, quoteID: number) {
     const xc = this.xcSection(host)
-    if (!xc) return false
+    if (!xc || !xc.marketRows) return false
```
Comment (JoeGruffins) [inline client/webserver/site/src/js/markets.ts:2231]: Could maybe use a comment as to why this is necessary so it doesn't get broken again in the future.

**Review by JoeGruffins — APPROVED** [review]
Wow, that's all it took.

File: client/webserver/site/src/js/markets.ts (hunk 1 above)
Comment (vctt94) [reply client/webserver/site/src/js/markets.ts:2233]: Thanks for the review Joe, I added the comment.

Comment (chappjc) [reply client/webserver/site/src/js/markets.ts:2233]: @JoeGruffins did you say you got an error on the wallets page too (as well as the markets page)?  It looks like @martonp fixed that in https://github.com/decred/dcrdex/commit/30e29d7107ac19ee8b9da5b4c701a18940ad898d#diff-1fb1a75f430311eb5c798880a2edbce5c4f529a823bd265bf255b3f4ba0cd8d9

**Review by chappjc — APPROVED** [review]

**Pull request merged**